pkg/executor/aggregate/agg_hash_final_worker.go

```diff
@@ -127,32 +118,30 @@ func (w *HashAggFinalWorker) loadFinalResult(sctx sessionctx.Context) {
 	if finished {
 		return
 	}
-	execStart := time.Now()
-	memSize := getGroupKeyMemUsage(w.groupKeys)
-	w.groupKeys = w.groupKeys[:0]
-	for groupKey := range w.groupSet.StringSet {
-		w.groupKeys = append(w.groupKeys, []byte(groupKey))
-	}
+
 	failpoint.Inject("ConsumeRandomPanic", nil)
-	w.memTracker.Consume(getGroupKeyMemUsage(w.groupKeys) - memSize)
-	partialResults := w.getPartialResult(sctx.GetSessionVars().StmtCtx, w.groupKeys, w.partialResultMap)
-	for i := 0; i < len(w.groupSet.StringSet); i++ {
-		for j, af := range w.aggFuncs {
-			if err := af.AppendFinalResult2Chunk(sctx, partialResults[i][j], result); err != nil {
-				logutil.BgLogger().Error("HashAggFinalWorker failed to append final result to Chunk", zap.Error(err))
-			}
-		}
-		if len(w.aggFuncs) == 0 {
-			result.SetNumVirtualRows(result.NumRows() + 1)
-		}
+
+	execStart := time.Now()
+	for _, results := range w.partialResultMap {
 		if result.IsFull() {
 			w.outputCh <- &AfFinalResult{chk: result, giveBackCh: w.finalResultHolderCh}
 			result, finished = w.receiveFinalResultHolder()
 			if finished {
 				return
 			}
 		}
+
+		for j, af := range w.aggFuncs {
+			if err := af.AppendFinalResult2Chunk(sctx, results[j], result); err != nil {
+				logutil.BgLogger().Error("HashAggFinalWorker failed to append final result to Chunk", zap.Error(err))
+			}
+		}
+
+		if len(w.aggFuncs) == 0 {
+			result.SetNumVirtualRows(result.NumRows() + 1)
+		}
```

Comment:
Why put this below the `if result.IsFull()` check

Reply:
> Why put this below the `if result.IsFull()` check

I think there is no difference whether this is put to the below of `if result.IsFull()`

Reply:
If there is no difference, we do not need to do this modification

Reply:
> If there is no difference, we do not need to do this modification

ok I will revoke it